core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -394,6 +394,8 @@ private boolean innerIndex(Index index) throws IOException {
                 final Translog.Location translogLocation = translog.add(new Translog.Index(index));
                 index.setTranslogLocation(translogLocation);
                 versionMap.putUnderLock(index.uid().bytes(), new VersionValue(updatedVersion, index.getTranslogLocation()));
+            } else {
+                versionMap.putUnderLock(index.uid().bytes(), new VersionValue(updatedVersion, null));
```

Comment:
can you put a comment here that we flush the version map after recovery so null is fine here...